azurerm/internal/services/synapse/tests/synapse_sql_pool_resource_test.go

```diff
@@ -0,0 +1,223 @@
+		},
+	})
+}
+
+func testCheckAzureRMSynapseSqlPoolExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Synapse.SqlPoolClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("synapse SqlPool not found: %s", resourceName)
+		}
+		id, err := parse.SynapseSqlPoolID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		if resp, err := client.Get(ctx, id.Workspace.ResourceGroup, id.Workspace.Name, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Synapse SqlPool %q does not exist", id.Name)
+			}
+			return fmt.Errorf("bad: Get on Synapse.SqlPoolClient: %+v", err)
+		}
+		return nil
+	}
+}
+
+func testCheckAzureRMsynapseSqlPoolDestroy(s *terraform.State) error {
+	client := acceptance.AzureProvider.Meta().(*clients.Client).Synapse.SqlPoolClient
+	ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_synapse_sql_pool" {
+			continue
+		}
+		id, err := parse.SynapseSqlPoolID(rs.Primary.ID)
+		if err != nil {
+			return err
+		}
+		if resp, err := client.Get(ctx, id.Workspace.ResourceGroup, id.Workspace.Name, id.Name); err != nil {
+			if !utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("bad: Get on Synapse.SqlPoolClient: %+v", err)
+			}
+		}
+		return nil
+	}
+	return nil
+}
+
+func testAccAzureRMSynapseSqlPool_basic(data acceptance.TestData) string {
+	template := testAccAzureRMSynapseSqlPool_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_synapse_sql_pool" "test" {
+  name                 = "acctestSP%s"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+}
+`, template, data.RandomString)
+}
+
+func testAccAzureRMSynapseSqlPool_requiresImport(data acceptance.TestData) string {
+	config := testAccAzureRMSynapseSqlPool_basic(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_synapse_sql_pool" "import" {
+  name                 = azurerm_synapse_sql_pool.test.name
+  synapse_workspace_id = azurerm_synapse_sql_pool.test.synapse_workspace_id
+}
+`, config)
+}
+
+func testAccAzureRMSynapseSqlPool_complete(data acceptance.TestData) string {
+	template := testAccAzureRMSynapseSqlPool_template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_synapse_sql_pool" "test" {
+  name                 = "acctestSP%s"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+  sku_name             = "DW500c"
+  create_mode          = "Default"
+  collation            = "SQL_Latin1_General_CP1_CI_AS"
+  data_encrypted       = true
+
+  tags = {
+    ENV = "Test"
+  }
+}
+`, template, data.RandomString)
+}
+
+func testAccAzureRMSynapseSqlPool_template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
```

Comment:
the provider block should be per-test (since this can vary on a per-test-case basis) - not in the template